lib/hocon/impl/tokenizer.rb

```diff
@@ -81,24 +105,39 @@ def create_whitespace_token_from_saver(base_origin, line_number)
       end
     end
 
-    # chars JSON allows a number to start with
-    FIRST_NUMBER_CHARS = "0123456789-"
-    # chars JSON allows to be part of a number
-    NUMBER_CHARS = "0123456789eE+-."
-    # chars that stop an unquoted string
-    NOT_IN_UNQUOTED_TEXT = "$\"{}[]:=,+#`^?!@*&\\"
+    def initialize(origin, input, allow_comments)
+      @origin = origin
+      @input = input
+      @allow_comments = allow_comments
+      @buffer = []
+      @line_number = 1
+      @line_origin = @origin.with_line_number(@line_number)
+      @tokens = []
+      @tokens << Tokens::START
+      @whitespace_saver = WhitespaceSaver.new
+    end
 
-    def self.problem(origin, what, message, suggest_quotes, cause)
-      if what.nil? || message.nil?
-        raise ConfigBugOrBrokenError.new("internal error, creating bad TokenizerProblemError")
+    # this should ONLY be called from nextCharSkippingComments
+    # or when inside a quoted string, or when parsing a sequence
+    # like ${ or +=, everything else should use
+    # nextCharSkippingComments().
+    def next_char_raw
+      if @buffer.empty?
+        begin
+          @input.readchar.chr
+        rescue EOFError
```

Comment:
Should we be rescuing an `IOError` here, and should we be throwing a config error?